scripts/base_network.py

```diff
@@ -680,6 +688,124 @@ def _adjust_capacities_of_under_construction_branches(n, config):
+    new_lines = _filter_assets(new_lines, allowed_statuses, ignored_projects)
+    new_links = _filter_assets(new_links, allowed_statuses, ignored_projects)
+
+    upg_buses = _filter_assets(upg_buses, allowed_statuses, ignored_projects)
+    upg_lines = _filter_assets(upg_lines, allowed_statuses, ignored_projects)
+    upg_links = _filter_assets(upg_links, allowed_statuses, ignored_projects)
+
+    # add 'commissioning_year' to existing assets (beginning of UNIX time)
+    if len(buses) > 0:
+        buses.loc[:, "commissioning_year"] = pd.to_datetime(0)
+    if len(lines) > 0:
+        lines.loc[:, "commissioning_year"] = pd.to_datetime(0)
+    if len(links) > 0:
+        links.loc[:, "commissioning_year"] = pd.to_datetime(0)
+
+    # append new assets
+    buses = pd.concat([buses, new_buses])
+    lines = pd.concat([lines, new_lines])
+    links = pd.concat([links, new_links])
+
+    # upgrade existing assets
+    buses = _add_upg_assets(buses, upg_buses)
+    lines = _add_upg_assets(lines, upg_lines)
+    links = _add_upg_assets(links, upg_links)
+
+    # Drop lines or links if their 'bus0' or 'bus1' columns contain
+    # buses that were dropped due to their 'tyndp_status'.
+    lines = lines.loc[lines.bus0.isin(buses.index) & lines.bus1.isin(buses.index)]
+    links = links.loc[links.bus0.isin(buses.index) & links.bus1.isin(buses.index)]
+
+    return buses, lines, links
+
+
+def _read_tyndp2020_buses(tyndp_file):
+    df = pd.read_csv(tyndp_file, index_col=0)
+    df.index = df.index.astype(str)
+    df.index.name = "bus_id"
+    df["carrier"] = df.pop("dc").map({True: "DC", False: "AC"})
+    df["commissioning_year"] = pd.to_datetime(df["commissioning_year"], format="%Y")
+    df["symbol"] = "substation"
+
+    return df
+
+
+def _read_tyndp2020_lines(tyndp_file):
+    df = pd.read_csv(tyndp_file, index_col=0, dtype={"bus0": "str", "bus1": "str"})
+    df.index = df.index.astype(str)
+    df.index.name = "line_id"
+    df["commissioning_year"] = pd.to_datetime(df["commissioning_year"], format="%Y")
+    df["v_nom"] = df["v_nom"].replace(np.nan, 220)
+    df["v_nom"] = df["v_nom"].astype(int)
+    df["num_parallel"] = 1
+    return df
+
+
+def _read_tyndp2020_links(tyndp_file):
+    df = pd.read_csv(tyndp_file, index_col=0, dtype={"bus0": "str", "bus1": "str"})
+    df.index = df.index.astype(str)
+    df.index.name = "link_id"
+    df["carrier"] = "DC"
+    df["commissioning_year"] = pd.to_datetime(df["commissioning_year"], format="%Y")
+    df = df.drop("v_nom", axis=1)
+    return df
+
+
+def _filter_assets(df, allowed_statuses, ignore_projects):
+    df = df.loc[df["tyndp_status"].isin(allowed_statuses)]
+    if ignore_projects:
+        df["project_id"] = _get_tyndp_project_id(df)
+        df = df.loc[~df.loc[:, "project_id"].isin(ignore_projects)]
+        df = df.drop("project_id", axis=1)
+
+    df = pd.concat(
+        [df, pd.DataFrame(index=df.index, columns=["under_construction"], data=True)],
+        axis=1,
+    )
```

Comment:
```suggestion
    df = df.assign(under_construction=True)
```
make hacky non-hacky